restli-server/src/main/java/com/linkedin/restli/server/config/ResourceMethodConfig.java

```diff
@@ -21,4 +21,9 @@
    * Config for whether this method will need to validate query parameters.
    */
   boolean shouldValidateQueryParams();
+
+  /**
+   * Config for whether this method will need to validate path/resource keys.
+   */
+  boolean shouldValidateResourceKey();
```

Comment:
nit: this should be pluralized, right? Based on my understanding, there can be multiple keys per method call (e.g. batch method)